### ISSUE TITLE
Alternative method for including DSL libs

### ISSUE THREAD

**Issue by LonelyCat124** [stfc/RegentParticleDSL#72]
This will be quite a big rewrite of some stuff but I discovered today that if I have 2 files:
```
import "regent"

function test1()
    require("test2")
end

test1()
test2()
```

and

```
import "regent"

function test2()
   print("Hello from test2")
end
```

This  worked correctly, 

My idea would be two reduce the DSL to only having 2 imports:
```
require("src/core_module")
---Create your parts structure etc. Can be via imports or not.
require("src/RegentParticleDSL")
```

The rest of the DSL imports could then be controlled by functions (with some default preset). For example
```
set_dimensionality(3)
set_periodic(false)
...
```

Once the setup is done (and we can also enable using an input file for these settings), then a simple call:
```
setup_dsl()
```
can sort all the headers etc. for the user.

This means an example script could (in the future) be something like
```
require("src/core_module")
require("mypart")
require("mykernels")
require("src/RegentParticleDSL")

read_config("path/to/file.yml")

setup_dsl()

task main()
[initialise(...)];

-- do timesteps

[output_result(...)];
end

regentlib.start(main)
```

I think is much cleaner than the current infrastructure, that can easily lead to headers of files like:
```
require("src/particles/core_part")
require("src/neighbour_search/cell_pair_tradequeues_nonperiod/import_2d_nonperiod")
require("defaults")
neighbour_init = require("src/neighbour_search/cell_pair_tradequeues_nonperiod/neighbour_init")
require("src/neighbour_search/cell_pair_tradequeues_nonperiod/neighbour_search")
require("src/neighbour_search/cell_pair_tradequeues_nonperiod/cell")
require("src/interactions/WC_SPH/timestep")
require("src/interactions/WC_SPH/import_WCSPH")
simple_hdf5_module = require("src/io_modules/HDF5/HDF5_simple_module")

local format = require("std/format")
local c = regentlib.c
local sqrtf = regentlib.sqrt(float)

variables = {}
variables.config = regentlib.newsymbol("config")
variables.particle_array = regentlib.newsymbol("particle_array")
```

I think we can even add functions for things like `import_WCSPH()` if someone wants to use the inbuilt modules in the future.

**Comment by LonelyCat124** [comment]
The one difficulty here that I'm not sure if it works or not (I will implement and see) is that the `neighbour_part` depends on the neighbour search, so is not necessarily correct at particle declaration time - but perhaps this will not matter as changing the reference to neighbour_part may work fine.

**Comment by LonelyCat124** [comment]
4292e1a301e1389a988c4e4e7a152574a52f6830 contains a first implementation of this. The testcases do not yet use this implementation, and i'll fix them next

The way this is setup is as follows right now (example use case):

```
import "regent"

require("src/RegentParticleDSL")
--OPTIONS CHOICES GO HERE
--Call setup part to get some initial headers. This name could be better
setup_part()
--Declare user particle type
fspace part{
  core_part_space : core_part,
  neighbour_part_space : neighbour_part,
  interactions : int
}

--Call setup_dsl at this point to add the rest of the stuff.
setup_dsl()

simple_hdf5_module = require("src/io_modules/HDF5/HDF5_simple_module")

local hdf5_read_mapper = {}
hdf5_read_mapper["cutoff"] = "core_part_space.cutoff"
hdf5_read_mapper["pos_x"] = "core_part_space.pos_x"
hdf5_read_mapper["pos_y"] = "core_part_space.pos_y"
hdf5_read_mapper["pos_z"] = "core_part_space.pos_z"

function symmetric_interaction_count_kernel(part1, part2, r2)
local kernel = rquote
  part1.interactions = part1.interactions + 1
  part2.interactions = part2.interactions + 1
end
return kernel
end

task main()

[simple_hdf5_module.initialisation("examples/interaction_count/basic_test.hdf5", hdf5_read_mapper, variables, 3.0, 3.0, 3.0)];
[neighbour_init.initialise(variables)];
[neighbour_init.update_cells(variables)];
[invoke(variables.config, {symmetric_interaction_count_kernel, SYMMETRIC_PAIRWISE}, BARRIER)];

end

regentlib.start(main)
```

**Comment by LonelyCat124** [comment]
Merged in #73
